st2common/st2common/models/db/stormbase.py

```diff
@@ -78,6 +78,23 @@ class StormBaseDB(StormFoundationDB):
     }
 
 
+class StormContentDB(StormBaseDB):
+    """Abstraction for a user content model from a pack."""
+    name = me.StringField(required=True)
+    pack = pack = me.StringField(
+        required=False,
+        help_text='Name of the content pack.',
+        unique_with='name')
+    file_uri = me.StringField(
```

Comment:
Per https://github.com/StackStorm/st2/blob/master/st2reactor/st2reactor/container/utils.py#L212 `artifact_uri` points to the sensor code file on disk and not location of the metadata file. Therefore, using the same name would not work. Even for senortypedb 2 different properties are required.